python/lsst/obs/base/cameraMapper.py

```diff
@@ -1081,49 +1140,28 @@ def _setFilter(self, mapping, item, dataId):
                 or isinstance(item, afwImage.ExposureF) or isinstance(item, afwImage.ExposureD)):
             return
 
-        try:
-            # getGen3Instrument returns class; need to construct it.
-            filterDefinitions = self.getGen3Instrument()().filterDefinitions
-        except NotImplementedError:
-            filterDefinitions = None
         itemFilter = item.getFilterLabel()  # may be None
         try:
             idFilter = mapping.need(['filter'], dataId)['filter']
         except dafPersist.NoResults:
             idFilter = None
 
-        if filterDefinitions is not None:
-            definitions = self._resolveFilters(filterDefinitions, idFilter, itemFilter)
-            self.log.debug("Matching filters for id=%r and label=%r are %s.",
-                           idFilter, itemFilter, definitions)
-            if len(definitions) == 1:
-                newLabel = list(definitions)[0].makeFilterLabel()
-                if newLabel != itemFilter:
-                    item.setFilterLabel(newLabel)
-            elif definitions:
-                self.log.warn("Multiple matches for filter %r with data ID %r.", itemFilter, idFilter)
-                # Can we at least add a band?
-                # Never expect multiple definitions with same physical filter.
-                bands = {d.band for d in definitions}  # None counts as separate result!
-                if len(bands) == 1 and itemFilter is None:
-                    band = list(bands)[0]
-                    item.setFilterLabel(afwImage.FilterLabel(band=band))
-            else:
-                # Unknown filter, nothing to be done.
-                self.log.warn("Cannot reconcile filter %r with data ID %r.", itemFilter, idFilter)
-        else:
-            if itemFilter is None:
-                # Old Filter cleanup, without the benefit of FilterDefinition
-                if self.filters is not None and idFilter in self.filters:
-                    idFilter = self.filters[idFilter]
-                try:
-                    # TODO: remove in DM-27177; at that point may not be able
-                    # to process IDs without FilterDefinition.
-                    with warnings.catch_warnings():
-                        warnings.filterwarnings("ignore", category=FutureWarning)
-                        item.setFilter(afwImage.Filter(idFilter))
-                except pexExcept.NotFoundError:
-                    self.log.warn("Filter %s not defined.  Set to UNKNOWN.", idFilter)
+        bestFilter = self._getBestFilter(itemFilter, idFilter)
+        if bestFilter is not None:
+            if bestFilter != itemFilter:
+                item.setFilterLabel(bestFilter)
```

Comment:
I think there's an `else` missing here? Otherwise, we fall into the next `elif` which could also be false and then nothing would get set at all.

Reply:
Intended behavior; I'll add a comment.

Reply:
Actually, wait... what are you expecting to be the behavior if `bestFilter == itemFilter`?

Reply:
If they are equal, then I would think you'd call `item.setFilterLabel(bestFilter)`? Oh, I see, in that case it's not necessary because FilterLabel already exists in the Exposure. A comment here to that effect would be great, please!